ui/lib/apps/SlowQuery/utils/useSlowQueryTableController.ts

```diff
@@ -137,37 +144,66 @@ export default function useSlowQueryTableController(
     [slowQueries, showFullSQL]
   )
 
-  useEffect(() => {
-    async function getSlowQueryList() {
+  const fetchSlowQueryList = useCallback(
+    async (force: boolean) => {
+      const {
+        schemas,
+        digest,
+        limit,
+        plans,
+        searchText,
+        timeRange,
+      } = queryOptions
+      const { desc, orderBy } = orderOptions
+      const { beginTime, endTime } = queryTimeRange
+      const cacheKey = `${schemas.join('_')}_${digest}_${limit}_${plans.join(
+        '_'
+      )}_${searchText}_${stringfyTimeRange(timeRange)}_${desc}_${orderBy}`
+      const cacheItem = slowQueryCacheMgr?.get(cacheKey)
```

Comment:
What if user just clicks refresh (and there is new SQL statements in the time range)?

Reply:
The refresh will trigger force fetch:

```
  function refresh() {
    setErrors([])
    fetchSlowQueryList(true)
}
```

Reply:
Cool

Reply:
Update the strategy by removing the cache when refreshing to simplify the code change and logic:

```js
  function refresh() {
    cacheMgr?.remove(cacheKey)

    setErrors([])
    setRefreshTimes((prev) => prev + 1)
}
```